core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -951,6 +951,25 @@ wasi_path_remove_directory(wasm_exec_env_t exec_env, wasi_fd_t fd,
     return wasmtime_ssp_path_remove_directory(curfds, fd, path, path_len);
 }
 
+static __wasi_timestamp_t
+get_timeout_for_poll_oneoff(const wasi_subscription_t *in)
+{
+    __wasi_timestamp_t timeout = 0;
+    const __wasi_subscription_t *clock_subscription = NULL;
+    if (clock_subscription == NULL
+        && (in->u.u.clock.flags & __WASI_SUBSCRIPTION_CLOCK_ABSTIME) == 0) {
+        clock_subscription = in;
+    }
+    if (clock_subscription != NULL) {
+        __wasi_timestamp_t ts = clock_subscription->u.u.clock.timeout / 1e9;
+        timeout = ts > INT_MAX ? -1 : (int)ts;
+    }
+    else {
+        timeout = -1;
```

Comment:
should we have an explicit cast here, since `timeout` is unsigned?

Reply:
done